### ISSUE TITLE
``enum.Enum`` type is blindly assigned to "Enum" of other modules

### ISSUE THREAD

**Issue by belm0** [PyCQA/pylint#5719]
### Bug description

For example, the prometheus_client module has an `Enum` class.  Pylint blindly assumes that references to `prometheus_client.Enum` are of type `enum.Enum`, and issues incorrect warnings.

```python
import prometheus_client

_foo_metric = prometheus_client.Enum('foo', 'help text', states=['bar', 'baz'])
_foo_metric.state('bar')  # pylint warning: Instance of 'foo' has no 'state' member
```

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
************* Module test
test.py:4:0: E1101: Instance of 'foo' has no 'state' member (no-member)
```

### Expected behavior

no warnings

### Pylint version

```shell
pylint 2.12.1
astroid 2.9.0
Python 3.8.12 (default, Sep 23 2021, 08:42:37)
[Clang 12.0.0 (clang-1200.0.32.29)]
```

### Additional dependencies

prometheus_client==0.8.0

**Comment by Pierre-Sassoulas** [comment]
We need to infer the name instead on relying on the name only.
